apps/gdalalg_raster_edit.cpp

```diff
@@ -55,6 +55,30 @@ GDALRasterEditAlgorithm::GDALRasterEditAlgorithm(bool standaloneStep)
 
     AddBBOXArg(&m_bbox);
 
+    {
+        auto &arg = AddArg("nodata", 0,
+                           _("Assign a specified nodata value to output bands "
+                             "('none' or numeric value)"),
+                           &m_nodata);
+        arg.AddValidationAction(
+            [this]()
+            {
+                if (m_nodata != "none" && m_nodata != "null")
+                {
+                    char *endptr = nullptr;
+                    CPLStrtod(m_nodata.c_str(), &endptr);
+                    if (endptr != m_nodata.c_str() + m_nodata.size())
+                    {
+                        ReportError(CE_Failure, CPLE_IllegalArg,
+                                    "Value of 'nodata' should be 'none' or a "
+                                    "numeric value");
```

Comment:
Some people like NaN, for some reason.

Reply:
this is accepted. Doc, error message and tests added